src/engine/thread_pool.h

```diff
@@ -57,7 +57,7 @@ class ThreadPool {
 
     /*! \brief Signal event upon destruction, even for exceptions (RAII) */
     struct SetReadyOnDestroy {
-      explicit inline SetReadyOnDestroy(std::shared_ptr<SimpleEvent> event)
+      explicit inline SetReadyOnDestroy(std::shared_ptr<SimpleEvent>& event)
```

Comment:
Bypass a superfluous locked increment/decrement when passing the shared pointer by value